core/referral/engine.go

```diff
@@ -221,6 +236,43 @@ func (e *Engine) RewardsFactorForParty(party types.PartyID) num.Decimal {
 	return num.DecimalZero()
 }
 
+func (e *Engine) RewardsFactorMultiplierAppliedForParty(party types.PartyID) num.Decimal {
+	return num.MinD(
+		e.RewardsFactorForParty(party).Mul(e.RewardsMultiplierForParty(party)),
+		e.rewardProportionUpdate,
+	)
+}
+
+func (e *Engine) RewardsMultiplierForParty(party types.PartyID) num.Decimal {
+	if e.programHasEnded {
+		return num.DecimalZero()
+	}
+
+	setID, isReferee := e.referees[party]
+	if !isReferee {
+		// This party is not eligible to referral program rewards.
+		return num.DecimalZero()
+	}
+
+	if e.isSetEligible(setID) != nil {
+		return num.DecimalZero()
+	}
+
+	balance, _ := e.staking.GetAvailableBalance(
+		string(e.sets[setID].Referrer.PartyID),
+	)
+
+	multiplier := num.DecimalOne()
+	for _, v := range e.currentProgram.StakingTiers {
```

Comment:
I don't remember, is it enforced to be sorted in the governance validation?

Reply:
no. good point

Reply:
added sorting in the updateProgram method